legacy-fabric-command-api-v2/1.6.4/src/main/java/net/legacyfabric/fabric/api/command/v2/lib/sponge/CommandResult.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2020 - 2024 Legacy Fabric
+ * Copyright (c) 2016 - 2022 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.legacyfabric.fabric.api.command.v2.lib.sponge;
+
+import java.util.Optional;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents the result of a command in Sponge.
+ */
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class CommandResult {
```

Comment:
some classes from SpongeAPI are fully abstracted from minecraft - see if we can move them into common instead of 1.6.4